lib/database_cleaner/moped/truncation_base.rb

```diff
@@ -23,9 +23,18 @@ def collections
           session.use(db)
         end
 
-        session['system.namespaces'].find(:name => { '$not' => /\.system\.|\$/ }).to_a.map do |collection|
-          _, name = collection['name'].split('.', 2)
-          name
+        if db_version.split('.').first.to_i >= 3
+          session.command(listCollections: 1)['cursor']['firstBatch'].map do |collection|
+            collection['name']
+          end
+          .reject do |collection_name|
+            collection_name =~ /\.system\.|\$/
```

Comment:
This regexp isn't matching `system` collections. It should be `collection_name =~ /.?system\.|\$/`

For MongoDB 3.x.x using Wired Tiger storage engine this PR is sufficient, because by default there are not `.system.` collections. But if you use MongoDB 3.x.x with MMapV1 storage engine, it will fail, because there still is the `system.indexes` collection.